src/aggregator/aggregator/generic_elem.go

```diff
@@ -537,9 +537,9 @@ func (e *GenericElem) processValueWithAggregationLock(
 			for _, point := range toFlush {
 				switch e.idPrefixSuffixType {
 				case NoPrefixNoSuffix:
-					flushLocalFn(nil, e.id, nil, point.TimeNanos, point.Value, e.sp)
+					flushLocalFn(nil, e.id, metric.GaugeType, nil, point.TimeNanos, point.Value, e.sp)
 				case WithPrefixWithSuffix:
-					flushLocalFn(e.FullPrefix(e.opts), e.id, e.TypeStringFor(e.aggTypesOpts, aggType),
+					flushLocalFn(e.FullPrefix(e.opts), e.id, metric.GaugeType, e.TypeStringFor(e.aggTypesOpts, aggType),
```

Comment:
Why is it a `Gauge`? Is the type not available here? If so, some comment might be useful.

Reply:
This is a base code for the code generation. By default, we set metric-type to Gauge (as most of them are) and just for Counters we replace it to Counter: https://github.com/m3db/m3/blob/gg/aggregator-keep-metric-type/src/aggregator/generated-source-files.mk#L16